task_sdk/src/airflow/sdk/definitions/asset/metadata.py

```diff
@@ -17,14 +17,34 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any
+from typing import (
+    Any,
+)
 
 import attrs
 
-from airflow.assets import AssetAlias, extract_event_key
+from airflow.sdk.definitions.asset import Asset, AssetAlias, _sanitize_uri
 
-if TYPE_CHECKING:
-    from airflow.assets import Asset
+__all__ = ["Metadata"]
+
+
+def extract_event_key(value: str | Asset | AssetAlias) -> str:
```

Comment:
I think this function should be exposed?

Reply:
Yep, we should. Let me update it